From m.kozlowski@tuxland.pl Fri Nov 17 08:49:11 2006
From: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
To: Lonnie Mendez <dignome@gmail.com>, Neil Whelchel <koyama@firstlight.net>
Subject: usb: cypress_m8 init error path fix
Date: Fri, 17 Nov 2006 17:49:22 +0100
Cc: Greg KH <greg@kroah.com>, linux-usb-devel@lists.sourceforge.net, linux-kernel@vger.kernel.org
Message-Id: <200611171749.22486.m.kozlowski@tuxland.pl>

If at some point cypress_init() fails deregister
only the resources that were registered until that point.

Signed-off-by: Mariusz Kozlowski <m.kozlowski@tuxland.pl>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/cypress_m8.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

--- gregkh-2.6.orig/drivers/usb/serial/cypress_m8.c
+++ gregkh-2.6/drivers/usb/serial/cypress_m8.c
@@ -1684,15 +1684,14 @@ static int __init cypress_init(void)
 
 	info(DRIVER_DESC " " DRIVER_VERSION);
 	return 0;
+
 failed_usb_register:
-	usb_deregister(&cypress_driver);
-failed_ca42v2_register:
 	usb_serial_deregister(&cypress_ca42v2_device);
-failed_hidcom_register:
+failed_ca42v2_register:
 	usb_serial_deregister(&cypress_hidcom_device);
-failed_em_register:
+failed_hidcom_register:
 	usb_serial_deregister(&cypress_earthmate_device);
-
+failed_em_register:
 	return retval;
 }
 
